detectors/aws/eks/go.mod

```diff
@@ -1,13 +1,13 @@
 module go.opentelemetry.io/contrib/detectors/aws/eks
 
-go 1.20
+go 1.21
```

Comment:
We still support Go 1.20. I see that the [compatibility tests](https://github.com/open-telemetry/opentelemetry-go-contrib/actions/runs/7992549988/job/21826297284?pr=5130) are passing. However, I am not sure if we should merge it.